### ISSUE TITLE
Error 404 on Heroku Instance when posting comments.

### ISSUE THREAD

**Issue by wini3d** [eduardoboucas/staticman#311]
I tried a heroku instance, the github account invite went **OK!** successfully, but having issues posting comments and getting a 500 error. I have already and only updated the form action to my heroku url.

- github account access token set to `admin:repo_hook` checked all under it.
- copy and pasted a generated `key.pem` file to key `RSA_PRIVATE_KEY` Heroku settings -> config vars (web/dashboard) 

@VincentTam I'm using your repo with the deploy branch. Any ideas?

**Comment by VincentTam** [comment]
Can you post

1. your form action URL (so that one can verify if you link is correct); and
2. the API server's response in JSON (for debug)?

You may find such info in the any modern web browsers's Web Developer Tools → network.

![sample output](https://user-images.githubusercontent.com/5748535/63198738-f1197a80-c07b-11e9-8aa7-54b94c1eff64.png)

Alternatively, an API client like Postman would show you the necessary info.

You can also check out [my list of Staticman errors](https://vincenttam.gitlab.io/post/2018-12-25-staticman-repo-setup-errors/) to know more about the causes of various Staticman errors.

**Comment by wini3d** [comment]
Ok see below..

You can also try it on my article [here](https://wini3d.github.io/2019-06-24-test-test/). I was using your staticman3/staticmanlab server earlier on this blog and just changed the form url.

I have not included or touched any recaptcha requirement ever since I have started working with staticman as to make things simple.

![stat](https://user-images.githubusercontent.com/12629729/64753109-9547f180-d554-11e9-82af-0cf06171231e.jpg)

**Comment by wini3d** [comment]
I have updated my `staticman.yml` and emptied the `siteKey` and `secret` and still getting 500 errors.

Never changed `allowedFields` ever since and it worked with the staticman3/staticmanlab

**Comment by VincentTam** [comment]
By saying "the API server's response in JSON", I mean sth like this.

```json
{
  "JSON": {
    "success": false,
    "rawError": {
      "message": "{\"message\":\"Not Found\",\"documentation_url\":\"https:\/\/developer.github.com\/v3\/git\/refs\/#create-a-reference\"}",
      "statusCode": 404
    },
    "errorCode": "GITHUB_CREATING_PR"
  },
  "Response payload": {
    "EDITOR_CONFIG": {
      "text": "{\"success\":false,\"rawError\":{\"message\":\"{\\\"message\\\":\\\"Not Found\\\",\\\"documentation_url\\\":\\\"https:\/\/developer.github.com\/v3\/git\/refs\/#create-a-reference\\\"}\",\"statusCode\":404},\"errorCode\":\"GITHUB_CREATING_PR\"}",
      "mode": "application\/json"
    }
  }
}
```

![Screenshot from 2019-09-12 06-35-15](https://user-images.githubusercontent.com/5748535/64754405-7e3eda80-d527-11e9-9330-38035235583d.png)


Plz consider providing a clearer screenshot by saving in PNG instead of JPG.

**Comment by wini3d** [comment]
Ok got it, so it's a 404 error and we're getting the same.

Here's is my full form action Url btw

https://staticmanx.herokuapp.com/v3/entry/github/wini3d/wini3d.github.io/master/comments

**Comment by wini3d** [comment]
@VincentTam  I did a fresh install with your deploy branch to this new staticmanx heroku app. Can you check out my staticman.yml and form if there's a problem there?

staticman.yml
https://github.com/wini3d/wini3d.github.io/blob/master/staticman.yml
form: 
https://github.com/wini3d/wini3d.github.io/blob/master/_includes/staticman-comments.html

**Comment by VincentTam** [comment]
Seems like this is relevant: https://github.com/eduardoboucas/staticman/issues/299#issuecomment-508199997.  I dunno whether my merging of #231 https://github.com/VincentTam/staticman/commit/f9675f91c5fa49cf1c0b48d498453072c41d96b5 caused the error.  It's sth that I did almost a year ago.  Excuse me I didn't remember that well.  It seems better to pick the commit before that for branching.  I'll check whether #231 is the culprit when I've got the time and motivation to do so.  Always fun testing and trying.

Remarks: Things might change on GitHub's side over time, as written in https://github.com/eduardoboucas/staticman/issues/217#issuecomment-405735386.

**Comment by wini3d** [comment]
Thanks @VincentTam , can you please link your recommended commit repo and branch directly? Sorry, getting confused with all these issue numbers.

Edit: I would assume this one with title 'Clean Up'. Is this ok?
https://github.com/VincentTam/staticman/tree/dd1969c1cd4fdecc1f159f37416d96cb4a1ad4ef

**Comment by wini3d** [comment]
I'm still getting errors with the before #231 commit. :/

@VincentTam Would you mind just cloning and sharing a copy of your heroku staticman3 instance as this would be easier and since you don't recall the one that works with heroku?

**Comment by wini3d** [comment]
Finally I got it working! =) Somehow, I had to give full access token (checked all) which I know is not recommended. I only checked `repo:invite` and `admin:repo_hook` before based on a [tutorial post](https://gist.github.com/ujjwal96/70eabafefa8c2e3f5fa900f352f16c5e) I read earlier.

Edit: found an [updated](https://gist.github.com/jannispaul/3787603317fc9bbb96e99c51fe169731) one that needs to set full repo access only

Thanks again for all the help @VincentTam :+1:

**Comment by VincentTam** [comment]
> Edit: found an [updated](https://gist.github.com/jannispaul/3787603317fc9bbb96e99c51fe169731) one that needs to set full repo access only

It's better to select a minimal set of priviledges due to the spam problem that I've mentioned.

![Screenshot_2019-09-12 Build software better, together](https://user-images.githubusercontent.com/5748535/64772141-662f8100-d550-11e9-90d9-588154bf32bb.png)

**Comment by wini3d** [comment]
Great thanks! Just curious @VincentTam , can you tell me exactly what `read:user` and `user:email` is for specifically with staticman usage?

**Comment by VincentTam** [comment]
I don't remember why I checked these one year ago.  They might be related to creating PR & accepting invitations.

**Comment by wini3d** [comment]
> I don't remember why I checked these one year ago. They might be related to creating PR & accepting invitations.

That's interesting. This is now my latest token that works.

![Personal Access Tokens](https://user-images.githubusercontent.com/12629729/64777316-7e24f600-d58c-11e9-9b5f-c2cefba504fd.png)

When I used only the `admin:repo_hook` from that first v2 tutorial, I failed once with the invite then added the `repo:invite` which I got to work next. 

With the public repo access, I'm now not sure if the repo_hook is still needed so might as well try it with another one.

In short, maybe this particular access token setup needs final confirmation so it can be properly documented just for staticman?

**Comment by VincentTam** [comment]
I've left a comment under your linked gist: https://gist.github.com/jannispaul/3787603317fc9bbb96e99c51fe169731.  @<span>jannispaul</span> has changed wordings to clarify the scopes.

I'm not sure if `admin:repo_hook` is relevant since I dunno node.js and I got Staticman's webhook working on GitHub with the legacy authentication method and v3 API without #288 and this scope.

As the name suggests, `public_repo` can't grant you rights to push PRs to private repos.  It's up to you to decide if your bot accepts invitations to private repos.

**Comment by wini3d** [comment]
> As the name suggests, public_repo can't grant you rights to push PRs to private repos. It's up to you to decide if your bot accepts invitations to private repos.

Yes, I am currently on a free plan so I have set it to public only. Thanks.

> I'm not sure if admin:repo_hook is relevant since I dunno node.js and I got Staticman's webhook working on GitHub with the legacy authentication method and v3 API without #288 and this scope.

I see, will try without this soon.
